panel/tests/ui/widgets/test_codeeditor.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+
+pytest.importorskip("playwright")
+
+from playwright.sync_api import expect
+
+from panel.tests.util import serve_component, wait_until
+from panel.widgets import CodeEditor
+
+pytestmark = pytest.mark.ui
+
+
+def test_code_editor(page):
+
+    editor = CodeEditor(value="print('Hello World!')")
+    serve_component(page, editor)
+    ace_input = page.locator(".ace_content")
+    expect(ace_input).to_have_count(1)
+    ace_input.click()
+
+    page.keyboard.press("Enter")
+    page.keyboard.type('print("Hello Panel!")')
+    wait_until(
+        lambda: page.locator(".ace_content").inner_text()
+        == "print('Hello World!')\nprint(\"Hello Panel!\")"
+    )
```

Comment:
expect(page.locator(".ace_content")).to_have_text("print('Hello World!')\nprint(\"Hello Panel!\")")

Reply:
Should have warned you, I didn't actually test this but you should always use the `expect` when checking page elements. `wait_until` is for testing if the Panel component has updated.